httpclient/src/main/java/org/apache/http/client/HttpResponseException.java

```diff
@@ -38,7 +38,7 @@
     private final int statusCode;
 
     public HttpResponseException(final int statusCode, final String s) {
-        super(s);
+        super(s == null || s.isEmpty() ? Integer.toString(statusCode) : s);
```

Comment:
@panchenko Looks good. Would you mind using `TextUtils#isBlank` here instead, please?

Reply:
@ok2c Please take a look